core/shared/src/main/scala/scalaz/zio/Schedule.scala

```diff
@@ -354,13 +354,14 @@ trait Schedule[-A, +B] extends Serializable { self =>
   /**
    * Applies random jitter to the schedule bounded by the factors 0.0 and 1.0.
    */
-  final def jittered: Schedule[A, B] = jittered(0.0, 1.0)
+  final def jittered(random: IO[Nothing, Double] = IO.sync(util.Random.nextDouble())): Schedule[A, B] =
```

Comment:
Let's do it this way:

```scala
final def jittered: ... = jittered(IO.sync(util.Random.nextDouble()))`

final def jittered(random: IO[Nothing, Double]): ... = jittered(0, 1, random)

final def jittered(min: Double, max: Double, random: IO[Nothing, Double]): ... 
```

This way it's backward-compatible.

Reply:
The first one can be a `val` I think

Reply:
Yes, although it will allocate memory which might not be used (it will only be used if the user calls `jittered`, so maybe best to be lazy in the allocation of that memory).